### DISCUSSION TITLE
Where is ViewModelBase?

### DISCUSSION THREAD

**Discussion by ghost** [AvaloniaUI/Avalonia#10432] [Q&A] [answered]
According to the last comment on [this issue](https://github.com/AvaloniaUI/Avalonia/issues/3239) ViewModelBase should be auto-generated, and it's definitely still advised to use it as per the docs [here](https://docs.avaloniaui.net/tutorials/todo-list-app/wiring-up-the-views), but in a brand new AvaloniaUI project ViewModelBase is nowhere to be found. I even decompiled the built DLL and it doesn't exist anywhere.

How do I get access to ViewModelBase?

**Comment by timunie** [comment, marked as answer]
It's only available in the MVVM template. In plain Avalonia, you'll not find it. For sure, you can implement it on your own and use any MVVM-library you prefer.
